### PR TITLE
Restore 'json' as default codec

### PR DESCRIPTION
**Pull request by andrewvc**
This restores the pre-refactor behavior

### CODE REVIEW

**Comment by andrewvc** [comment]
This will fix #35

File: lib/logstash/outputs/rabbitmq.rb [hunk 1]
```diff
@@ -33,15 +36,19 @@ class RabbitMQ < LogStash::Outputs::Base
       # Should RabbitMQ persist messages to disk?
       config :persistent, :validate => :boolean, :default => true
 
+      def initialize(params)
+        params["codec"] = "json" if !params["codec"]
+        super
+      end
+
```
Comment (ph) [inline lib/logstash/outputs/rabbitmq.rb]: This is the same as defining default :codec, "json" in the plugin?

Comment (andrewvc) [inline lib/logstash/outputs/rabbitmq.rb]: I didn't know you could do that! (I cribbed that code from the old plugin code). Will fix!

Comment (untergeek) [inline lib/logstash/outputs/rabbitmq.rb]: Indeed.  This seems like it should just be handled in the config area.

**Comment by andrewvc** [comment]
P.S. In testing this you'll want to start up a rabbitmq server and then run `bin/rspec spec --tag integration`

File: lib/logstash/outputs/rabbitmq.rb [hunk 2]
```diff
@@ -15,6 +18,8 @@ class RabbitMQ < LogStash::Outputs::Base
       include LogStash::PluginMixins::RabbitMQConnection
 
       config_name "rabbitmq"
+      
+      default :codec, "json"
```
Comment (untergeek) [inline lib/logstash/outputs/rabbitmq.rb]: Should that read more like:

```
  config :codec, :default => "json"
```

See the [file input](https://github.com/logstash-plugins/logstash-input-file/blob/master/lib/logstash/inputs/file.rb#L67-L69) for an example.

Comment (untergeek) [inline lib/logstash/outputs/rabbitmq.rb]: That allows documentation to be associated with the setting, which is probably a good thing here.

Comment (andrewvc) [inline lib/logstash/outputs/rabbitmq.rb]: Ah, the file input does it the way I do it here. Do you just mean I should add comments above?

**Comment by ph** [comment]
lgtm test passes

File: lib/logstash/outputs/rabbitmq.rb (hunk 2 above)
Comment (andrewvc) [inline lib/logstash/outputs/rabbitmq.rb]: @untergeek I added some comments. Looking better?

**Comment by andrewvc** [comment]
Thanks @ph !

File: lib/logstash/outputs/rabbitmq.rb [hunk 3]
```diff
@@ -16,6 +16,9 @@ class RabbitMQ < LogStash::Outputs::Base
 
       config_name "rabbitmq"
 
+      # The default codec for this plugin is JSON. You can override this to suit your particular needs however.
```
Comment (untergeek) [inline lib/logstash/outputs/rabbitmq.rb]: This is great!

**Pull request merged**